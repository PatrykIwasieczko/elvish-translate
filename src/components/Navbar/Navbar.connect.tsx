import { login, logout } from "../../store/actions";
import { RootState, useAppDispatch, useAppSelector } from "../../store/store";
import { Navbar } from "./Navbar";

export const NavbarConnect = () => {
  const dispatch = useAppDispatch();
  const isAuthenticated = useAppSelector(
    (state: RootState) => state.isAuthenticated
  );

  return (
    <Navbar
      isAuthenticated={isAuthenticated}
      login={() => dispatch(login())}
      logout={() => dispatch(logout())}
    />
  );
};
