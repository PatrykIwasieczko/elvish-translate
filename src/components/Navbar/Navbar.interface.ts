export type NavbarProps = {
  isAuthenticated: boolean;
  login: () => void;
  logout: () => void;
};
