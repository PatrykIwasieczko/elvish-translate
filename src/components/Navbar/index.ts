export { Navbar } from "./Navbar";
export { NavbarConnect } from "./Navbar.connect";
