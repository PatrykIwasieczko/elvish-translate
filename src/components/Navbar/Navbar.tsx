import { Link } from "react-router-dom";
import { NavbarProps } from "./Navbar.interface";

export const Navbar: React.FC<NavbarProps> = ({
  isAuthenticated,
  login,
  logout,
}) => (
  <nav className="bg-gray-800 p-4">
    <div className="container mx-auto flex justify-between items-center">
      <Link to="/" className="text-white text-xl">
        Elvish Translator
      </Link>
      <div className="flex items-center gap-4">
        <Link to="/favourites" className="text-white">
          Favourites
        </Link>
        {isAuthenticated ? (
          <button className="text-white" onClick={logout}>
            Logout
          </button>
        ) : (
          <button className="text-white" onClick={login}>
            Login
          </button>
        )}
      </div>
    </div>
  </nav>
);
