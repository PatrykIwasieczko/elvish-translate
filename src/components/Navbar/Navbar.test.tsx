import { render } from "@testing-library/react";
import { Navbar } from "./Navbar";
import { MemoryRouter } from "react-router-dom";

describe("components/Navbar", () => {
  it("should render Navbar component when logged in", () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar isAuthenticated login={jest.fn()} logout={jest.fn()} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render Navbar component when logged out", () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar isAuthenticated={false} login={jest.fn()} logout={jest.fn()} />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
