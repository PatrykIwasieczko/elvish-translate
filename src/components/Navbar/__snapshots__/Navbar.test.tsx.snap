// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navbar should render Navbar component when logged in 1`] = `
<div>
  <nav
    class="bg-gray-800 p-4"
  >
    <div
      class="container mx-auto flex justify-between items-center"
    >
      <a
        class="text-white text-xl"
        href="/"
      >
        Elvish Translator
      </a>
      <div
        class="flex items-center gap-4"
      >
        <a
          class="text-white"
          href="/favourites"
        >
          Favourites
        </a>
        <button
          class="text-white"
        >
          Logout
        </button>
      </div>
    </div>
  </nav>
</div>
`;

exports[`components/Navbar should render Navbar component when logged out 1`] = `
<div>
  <nav
    class="bg-gray-800 p-4"
  >
    <div
      class="container mx-auto flex justify-between items-center"
    >
      <a
        class="text-white text-xl"
        href="/"
      >
        Elvish Translator
      </a>
      <div
        class="flex items-center gap-4"
      >
        <a
          class="text-white"
          href="/favourites"
        >
          Favourites
        </a>
        <button
          class="text-white"
        >
          Login
        </button>
      </div>
    </div>
  </nav>
</div>
`;
